lib/koine/repository/id_aware_entity.rb

```diff
@@ -0,0 +1,26 @@
+module Koine
+  module Repository
+    module IdAwareEntity
+      def find(id)
+        data = adapter.find_one_by(id: id)
+        return data unless data
+        entity = new_entity
+        hydrate(data, entity)
+        entity
+      end
+
+      def create(entity)
+        values = hydrator.extract(entity)
+        values.delete('id')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.